components/odh-notebook-controller/controllers/notebook_runtime_test.go

```diff
@@ -317,6 +317,101 @@ var _ = Describe("Runtime images ConfigMap should be mounted", func() {
 					},
 				},
 			},
+			{
+				name:         "ImageStream with formatKeyName edge cases",
+				notebookName: "test-notebook-runtime-format-key",
+				expectedConfigMapData: map[string]string{
+					"foo-bar.json":       `{"display_name":"foo  bar","metadata":{"display_name":"foo  bar","image_name":"quay.io/opendatahub/test1","pull_policy":"IfNotPresent","tags":["tag1"]},"schema_name":"runtime-image"}`,
+					"invalid-chars.json": `{"display_name":" !@#$ invalid chars","metadata":{"display_name":" !@#$ invalid chars","image_name":"quay.io/opendatahub/test2","pull_policy":"IfNotPresent","tags":["tag2"]},"schema_name":"runtime-image"}`,
+					"cz.json":            `{"display_name":"CZ ěščřžýáíé","metadata":{"display_name":"CZ ěščřžýáíé","image_name":"quay.io/opendatahub/test3","pull_policy":"IfNotPresent","tags":["tag3"]},"schema_name":"runtime-image"}`,
+				},
+				imageStream: &imagev1.ImageStream{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "ImageStream",
+						APIVersion: "image.openshift.io/v1",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "format-key-test-image",
+						Namespace: "redhat-ods-applications",
+						Labels: map[string]string{
+							"opendatahub.io/runtime-image": "true",
+						},
+					},
+					Spec: imagev1.ImageStreamSpec{
+						LookupPolicy: imagev1.ImageLookupPolicy{
+							Local: true,
+						},
+						Tags: []imagev1.TagReference{
+							{
+								Name: "tag1",
+								Annotations: map[string]string{
+									"opendatahub.io/runtime-image-metadata": `
+										[
+											{
+												"display_name": "foo  bar",
+												"metadata": {
+													"tags": ["tag1"],
+													"display_name": "foo  bar",
+													"pull_policy": "IfNotPresent"
+												},
+												"schema_name": "runtime-image"
+											}
+										]
+									`,
+								},
+								From: &corev1.ObjectReference{
+									Kind: "DockerImage",
+									Name: "quay.io/opendatahub/test1",
+								},
+							},
+							{
+								Name: "tag2",
+								Annotations: map[string]string{
+									"opendatahub.io/runtime-image-metadata": `
+										[
+											{
+												"display_name": " !@#$ invalid chars",
```

Comment:
let's include also `|` character as we use this one explicitly in our manifests, please